packages/context/src/inject.ts

```diff
@@ -52,6 +53,14 @@ export interface InjectionMetadata extends ResolutionOptions {
    * It's usually set by the decorator implementation.
    */
   decorator?: string;
+  /**
+   * Control if the dependency is optional, default to false
+   */
+  optional?: boolean;
```

Comment:
I believe the flag `optional` is already inherited from `ResolutionOptions`. Can you remove the duplicate definition from `InjectionMetadata` please?